subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -509,6 +509,9 @@ static int run_test(struct ztest_suite_node *suite, struct ztest_unit_test *test
 {
 	int ret = TC_PASS;
 
+#if CONFIG_ZTEST_TEST_DELAY_MS > 0
+	k_busy_wait(CONFIG_ZTEST_TEST_DELAY_MS * 1000);
```

Comment:
Sorry just noticed this as well: should probably use `USEC_PER_MSEC` instead of `1000`.